lib/src/default_index_store.rs

```diff
@@ -1109,10 +1109,13 @@ impl<'a> CompositeIndex<'a> {
     }
 
     /// Parents before children
-    fn topo_order(&self, input: &mut dyn Iterator<Item = &CommitId>) -> Vec<IndexEntry<'a>> {
+    fn topo_order(&self, input: &mut dyn Iterator<Item = &CommitId>) -> Vec<CommitId> {
         let mut entries_by_generation = input.map(|id| self.entry_by_id(id).unwrap()).collect_vec();
         entries_by_generation.sort_unstable_by_key(|e| e.pos);
         entries_by_generation
+            .into_iter()
+            .map(|entry| entry.commit_id())
+            .collect()
```

Comment:
nit: maybe it's simpler to collect `Vec<CommitId>` first, and sort them by sort_by_cached_key()?

Reply:
Yes, that turned out much simpler, especially since we have `commit_id_to_pos()` so we don't even need to create the `IndexEntry`.